providers/reddit/reddit_test.go

```diff
@@ -0,0 +1,28 @@
+package reddit
+
+import (
+	"fmt"
+	"testing"
+)
+
+// TestBuildURI makes sure the BuildURI method builds valid URIs.
+func TestBuildURI(t *testing.T) {
+	p := Provider{}
+	cases := []struct {
+		query, want string
+	}{
+		// Regular reddit search.
+		{"best startups",
+			fmt.Sprintf("https://www.reddit.com/search?q=%s", "best+startups")},
+		// Subreddit search.
+		{"/r/cscareerquestions best startups",
```

Comment:
Needs one more test case, need to test only the subreddit being passed. I think if they only pass a subreddit and no search, that we return an empty URI. I can then catch empty URI's coming from the provider and show a good error message.